src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/UnprotectedAttributeTests.cs

```diff
@@ -305,8 +305,28 @@ public static void TestUnprotectedAttributes1_EmptySet()
 
             EnvelopedCms ecms = new EnvelopedCms();
             ecms.Decode(encodedMessage);
+
+            Assert.Equal(2, ecms.UnprotectedAttributes.Count);
+            
+            CryptographicAttributeObjectCollection collection = ecms.UnprotectedAttributes;
+
+            int countOfAttr0 = collection[0].Values.Count;
+            int countOfAttr1 = collection[1].Values.Count;
+
+            Assert.True(countOfAttr0 == 0 || countOfAttr0 == 0);
+
+            CryptographicAttributeObject emptyAttributeObj = (countOfAttr0 == 0) ?
+                collection[0] :
+                collection[1];
+
+            CryptographicAttributeObject nonEmptyAttributeObj = (countOfAttr0 != 0) ?
+                collection[0] :
+                collection[1];
             
-            Assert.Equal(1, ecms.UnprotectedAttributes.Count);
+            Assert.Equal(2, nonEmptyAttributeObj.Values.Count);
+
+            Assert.Equal(Oids.DocumentName, emptyAttributeObj.Oid.Value);
+            Assert.Equal(Oids.DocumentDescription, nonEmptyAttributeObj.Oid.Value);
```

Comment:
Rather than asserting the OID to the nested value counts, why not go the other way?  Look for the DocumentName entry, assert it has 0 attribute values.